services/contributions.js

```diff
@@ -30,15 +30,17 @@ const getUserContributions = async (username) => {
 
       noteworthyObject.task = extractTaskdetails(task)
 
-      for (const userId of task.participants) {
-        const username = await userUtils.getUsername(userId)
-        const userDetails = participantsDetailsMap.get(username)
-        if (userDetails) {
-          participantsDetails.push(userDetails)
-        } else {
-          const user = await getUserDetails(username)
-          participantsDetailsMap.set(username, user)
-          participantsDetails.push(user)
+      if (Array.isArray(task.participants)) {
```

Comment:
We are creating an extra `if` block to increase the complexity of a larger piece of code.

This may be more predicatble:

```js
const { participants = [] } = task
for (participants) { ... } 
```

Reply:
Sorry for the trouble, The changes are not from the current PR